sources/org.osbuild.files

```diff
@@ -71,8 +83,18 @@ def main(options, checksums, cache, output):
 
     with concurrent.futures.ProcessPoolExecutor(max_workers=15) as executor:
         requested_urls = []
+        rhsm_secrets = None
+
         for checksum in checksums:
             try:
+                secrets = urls[checksum].get("secrets")
```

Comment:
To be backwards compatible, we need to first check if this is a dict and otherwise transform the plain string into a dict with just the `url` member so that `fetch` always gets the same object (or we need another `if` there as well).

Reply:
(This is your `AttributeError: 'str' object has no attribute 'get'` in the failed tests)

Reply:
This is fixed